whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/importer/DeltaImporter.java

```diff
@@ -0,0 +1,175 @@
+        }
+    }
+
+    private void applyDeltaCreation(final MirrorDeltaInfo deltaInfo) {
+        final RpslObjectUpdateInfo rpslObjectCretedInfo = nrtm4ClientRepository.persistRpslObject(deltaInfo.getRpslObject());
+        nrtm4ClientRepository.createIndexes(deltaInfo.getRpslObject(), rpslObjectCretedInfo);
+    }
+
+    private void applyDeltaDeletion(final MirrorDeltaInfo deltaInfo) {
+        final RpslObjectUpdateInfo rpslObjectUpdateInfo = nrtm4ClientRepository.getMirroredObjectId(deltaInfo.getObjectType(), deltaInfo.getPrimaryKey());
+
+        if (serialDoesNotExist(rpslObjectUpdateInfo)){
+            LOGGER.error("delta with pkey: {} not deleted because serial doesn't exist", deltaInfo.getPrimaryKey());
+            return;
+        }
+        nrtm4ClientRepository.removeMirroredObjectAndUpdateSerials(rpslObjectUpdateInfo);
+    }
+
+    private void applyDeltaUpdate(final MirrorDeltaInfo deltaInfo, final RpslObjectUpdateInfo rpslObjectUpdateInfo) {
+        if (serialDoesNotExist(rpslObjectUpdateInfo)) {
+            LOGGER.error("delta with pkey: {} not updated because serial doesn't exist", deltaInfo.getPrimaryKey());
+            return;
+        }
+        nrtm4ClientRepository.updateMirroredObject(deltaInfo.getRpslObject(), rpslObjectUpdateInfo);
+    }
+
+    private boolean serialDoesNotExist(final RpslObjectUpdateInfo rpslObjectUpdateInfo) {
+        if (rpslObjectUpdateInfo != null){
+            return nrtm4ClientRepository
+                    .getSerialByObjectId(rpslObjectUpdateInfo.getObjectId(), rpslObjectUpdateInfo.getSequenceId()) == null;
+        }
+        return false;
+    }
+
+    private List<UpdateNotificationFileResponse.NrtmFileLink> getNewDeltas(String source, UpdateNotificationFileResponse updateNotificationFile) {
+        final NrtmClientVersionInfo nrtmClientVersionInfo = nrtm4ClientInfoRepository.getNrtmLastVersionInfoForDeltasPerSource(source);
+
+        if (nrtmClientVersionInfo == null){
+            return updateNotificationFile.getDeltas();
+        }
+
+        return updateNotificationFile.getDeltas()
+                .stream()
+                .filter(delta -> delta.getVersion() > nrtmClientVersionInfo.version())
+                .toList();
+    }
+
+
+    private static List<MirrorDeltaInfo> getMirrorDeltaObjects(final String[] records) {
+        final List<MirrorDeltaInfo> mirrorDeltaInfos = Lists.newArrayList();
+        for (int i = 1; i < records.length; i++) {
+            final JSONObject jsonObject = new JSONObject(records[i]);
+            final String deltaAction = jsonObject.getString("action");
+            final String deltaObjectType = jsonObject.optString("object_class", null);
+            final String deltaPrimaryKey = jsonObject.optString("primary_key", null);
+            final String deltaUpdatedObject = jsonObject.optString("object", null);
+            final RpslObject rpslObject = !StringUtil.isNullOrEmpty(deltaUpdatedObject) ?
```

Comment:
Is null allowed? For a delete I guess?

Reply:
check tests json
`{"action":"delete","object_class":"DOMAIN","primary_key":"7.4.1.8.0.1.a.2.ip6.arpa"}`
delete doesn't have "object"
so yes, the delete doesn't allow it